sparse/unit_test/Test_Sparse_sptrsv.hpp

```diff
@@ -38,1111 +38,557 @@ using namespace KokkosKernels;
-      size_type nsupers = 4;
-      Kokkos::View<int *, Kokkos::HostSpace> supercols("supercols",
-                                                       1 + nsupers);
-      supercols(0) = 0;
-      supercols(1) = 2;     // two columns
-      supercols(2) = 3;     // one column
-      supercols(3) = 4;     // one column
-      supercols(4) = 5;     // one column
-      int *etree   = NULL;  // we generate graph internally
-
-      // invert diagonal blocks
-      bool invert_diag = true;
-      khL.set_sptrsv_invert_diagonal(invert_diag);
-      khU.set_sptrsv_invert_diagonal(invert_diag);
-
-      // > symbolic (on host)
-      sptrsv_supernodal_symbolic(nsupers, supercols.data(), etree, L.graph,
-                                 &khL, U.graph, &khU);
-      // > numeric (on host)
-      sptrsv_compute(&khL, L);
-      sptrsv_compute(&khU, U);
-      Kokkos::fence();
-
-      // > solve
-      ValuesType b("b", nrows);
-      Kokkos::deep_copy(b, B);
-      Kokkos::deep_copy(X, ZERO);
-      sptrsv_solve(&khL, &khU, X, b);
-      Kokkos::fence();
-
-      // > check
-      scalar_t sum = 0.0;
-      Kokkos::parallel_reduce(
-          Kokkos::RangePolicy<typename device::execution_space>(0, X.extent(0)),
-          ReductionCheck<ValuesType, scalar_t, lno_t>(X), sum);
-      if (sum != lhs.extent(0)) {
-        std::cout << "Supernode Tri Solve FAILURE : " << sum << " vs."
-                  << lhs.extent(0) << std::endl;
-        khL.get_sptrsv_handle()->print_algorithm();
-      } else {
-        std::cout << "Supernode Tri Solve SUCCESS" << std::endl;
-        khL.get_sptrsv_handle()->print_algorithm();
-      }
-      EXPECT_TRUE(sum == scalar_t(X.extent(0)));
+      Crs triMtx("triMtx", nrows, nrows, nnz, values, row_map, entries);
+      KokkosSparse::spmv("N", ONE, triMtx, known_lhs, ZERO, rhs);
 
-      khL.destroy_sptrsv_handle();
-      khU.destroy_sptrsv_handle();
-    }
+      {
+        KernelHandle kh;
+        bool is_lower_tri = false;
+        kh.create_sptrsv_handle(SPTRSVAlgorithm::SEQLVLSCHD_TP1, nrows,
+                                is_lower_tri);
+
+        sptrsv_symbolic(&kh, row_map, entries);
+        Kokkos::fence();
+
+        sptrsv_solve(&kh, row_map, entries, values, rhs, lhs);
+        Kokkos::fence();
+
+        scalar_t sum = 0.0;
+        Kokkos::parallel_reduce(range_policy_t(0, lhs.extent(0)),
+                                ReductionCheck(lhs), sum);
+        EXPECT_EQ(sum, lhs.extent(0));
+
+        Kokkos::deep_copy(lhs, ZERO);
+        kh.get_sptrsv_handle()->set_algorithm(SPTRSVAlgorithm::SEQLVLSCHD_RP);
+        sptrsv_solve(&kh, row_map, entries, values, rhs, lhs);
+        Kokkos::fence();
+
+        sum = 0.0;
+        Kokkos::parallel_reduce(range_policy_t(0, lhs.extent(0)),
+                                ReductionCheck(lhs), sum);
+        EXPECT_EQ(sum, lhs.extent(0));
+
+        // FIXME Issues with various integral type combos - algorithm currently
```

Comment:
What is this :)

Reply:
Not sure, this was there before I touched the file.